File: dsa/week(1)/pointer.js
let i = 10;
let p = { value: i }; // Simulate a pointer-like behavior using an object
console.log(p);

// In JavaScript, you can't directly do pointer arithmetic.
// However, for demonstration, we use a simple offset simulation:
p.offset = 2; // Simulate p = p + 2
console.log(`Pointer offset: ${p.offset}`);

p.offset -= 2; // Simulate p = p - 2
console.log(`Pointer reset offset: ${p.offset}`);

let d = 102.3;
let dp = { value: d }; // Simulate pointer behavior for a double
console.log(dp);

// Simulate dp++
dp.increment = 1;
console.log(`Pointer increment: ${dp.increment}`);

File: dsa/week(1)/pointer.py
i = 10
p = [i]  # Use a list to simulate pointer-like behavior
print(p)

# Pointer arithmetic doesn't exist in Python, but we can mimic the logic
p_offset = 2  # Simulate p = p + 2
print(f"Pointer offset: {p_offset}")

p_offset -= 2  # Simulate p = p - 2
print(f"Pointer reset offset: {p_offset}")

d = 102.3
dp = [d]  # Simulate pointer behavior for a double
print(dp)

# Simulate dp++
dp_offset = 1
print(f"Pointer increment: {dp_offset}")
